llvm/include/llvm/CAS/CachingOnDiskFileSystem.h

```diff
@@ -45,6 +45,11 @@ class CachingOnDiskFileSystem : public ThreadSafeFileSystem {
   /// accesses are only recorded in the currently active scope.
   virtual void trackNewAccesses() = 0;
 
+  /// Exclude \p Path from the current tracking scope. If \p Path is a
+  /// directory, this excludes the contents of that directory as well. \returns
+  /// any error seen resolving \p Path.
+  virtual std::error_code excludeFromTracking(const Twine &Path) = 0;
```

Comment:
Also don't like this API for the same reason. What is the reason behind tracking module cache directory?

Is it module file that is changing? Or the module directory is remapped?

Reply:
We talked about this offline:
* exists() does **not** have an anti-dependency issue, because we still track the file, it just has empty contents.
* excludeFromTracking has a potential anti-dependency issue if you misuse it. It's not a problem for the module cache, because we don't use the module cache in the cached build, only the scanner.  I will improve the doc comments here to mention this is a potential issue if you use it on a different path.
* The module cache only contains files used as implementation details of the scanner, we don't use any of these during the compilation itself.
* If we changed the scanner to no longer write pcm files, we might no longer need to exclude it.
* I think having a way to exclude paths from tracking is still a generally useful thing to have.  It's certainly something you need to be careful with though.